### PR TITLE
Disable sticky scroll in diff editor when collapse unmodified code is enabled

### PR DESCRIPTION
**Pull request by hediet**

### CODE REVIEW

File: src/vs/editor/browser/widget/diffEditorWidget2/utils.ts [hunk 1]
```diff
@@ -170,3 +170,19 @@ export function animatedObservable(base: IObservable<number, boolean>, store: Di
 function easeOutExpo(t: number, b: number, c: number, d: number): number {
 	return t === d ? b + c : c * (-Math.pow(2, -10 * t / d) + 1) + b;
 }
+
+export function deepMerge<T extends {}>(source1: T, source2: Partial<T>): T {
```
Comment (TylerLeonhardt) [inline src/vs/editor/browser/widget/diffEditorWidget2/utils.ts:174]: This seems similar to https://github.com/microsoft/vscode/blob/main/src/vs/base/common/objects.ts#L89

Maybe use that or a deepClone and then mixin if you want a truly new object.

File: src/vs/editor/browser/widget/diffEditorWidget2/utils.ts [hunk 2]
```diff
@@ -170,3 +170,19 @@ export function animatedObservable(base: IObservable<number, boolean>, store: Di
...
+	const result = {} as T;
+	for (const key in source1) {
```
Comment (TylerLeonhardt) [inline src/vs/editor/browser/widget/diffEditorWidget2/utils.ts:176]: Are you trying to return a completely new object? If so, I don't think this is properly cloning source1 if it has any nested properties.

File: src/vs/editor/browser/widget/diffEditorWidget2/utils.ts (hunk 1 above)
Comment (hediet) [reply src/vs/editor/browser/widget/diffEditorWidget2/utils.ts:174]: mixin uses `any` types...

File: src/vs/editor/browser/widget/diffEditorWidget2/utils.ts (hunk 2 above)
Comment (hediet) [reply src/vs/editor/browser/widget/diffEditorWidget2/utils.ts:176]: I'm not trying to return a completely new object, but I'm not touching the original objects. This is also what makes this method much more efficient as deepClone + mixin.

**Review by TylerLeonhardt — APPROVED** [review]

**Pull request merged**